### ISSUE TITLE
Errors in deserializeBinary causes uncaught exception

### ISSUE THREAD

**Issue by tlrobinson** [improbable-eng/grpc-web#829]
**Versions of relevant software used**

0.13.0

**What happened**

A corrupted gRPC response can lead to an uncaught exception, which will crash React Native applications.

It seems this happens because errors in `deserializeBinary` are not caught here: https://github.com/improbable-eng/grpc-web/blob/master/client/grpc-web/src/client.ts#L128

**What you expected to happen**

Errors in `deserializeBinary` should be caught and returned via `rawOnError`.

**How to reproduce it (as minimally and precisely as possible)**:

The easiest way I found is to modify your generated code to throw an exception inside a `deserializeBinaryFromReader` method, for example. `throw new Error()` should be sufficient.

**Comment by johanbrandhorst** [comment]
I don't think we control the implementation of `deserializeBinaryFromReader`, I think that comes from the generated javascript produced by `protoc`. Do you think it would be appropriate for us to catch around the call itself?
